biocrnpyler/sbmlutil.py

```diff
@@ -121,197 +159,92 @@ def find_parameter(mixture, id):
-                "'s1':species (chemical_reaction_network.species)"
-                "'d':species (chemical_reaction_network.species), ")
+        #Reversible reactions are always seperated into two seperate reactions
+        if r.is_reversible:
+            rxn_id = f'r{rxn_count}rev'
+            add_reaction(model=model, crn_reaction=r, reaction_id=rxn_id, stochastic=stochastic, reverse_reaction = True, **kwargs)
 
-        s = str(propensity_params['s1']).replace("'", "")
-        d = str(propensity_params['d']).replace("'", "")
-        s_species_id = getSpeciesByName(model,s).getId()
-        d_species_id = getSpeciesByName(model,d).getId()
 
-        ratestring+=f"*{d_species_id}*{s_species_id}^n/({s_species_id}^n + K)"
+def add_reaction(model, crn_reaction, reaction_id: str, stochastic: bool=False, reverse_reaction : bool=False, **kwargs):
+    """adds a sbml_reaction to an sbml model
 
-        annotation_dict["s1"] = s_species_id
-        annotation_dict["d"] = d_species_id
+    :param model: an sbml model created by create_sbml_model()
+    :param crn_reaction: must be a chemical_reaction_network.reaction object
+    :param reaction_id: unique id of the reaction
+    :param stochastic: stochastic model flag
+    :return: SBML sbml_reaction object
+    """
 
-    elif propensity_type == "proportionalhillnegative":
-        if not ("s1" in propensity_params and "d" in propensity_params):
-            raise ValueError("proportionalhillnegative propensities, "
-                "p(s1, d; k, K, n) = k*d/(s1^n + K), require the following key in the propensity_params dictionary:"
-                "'s1':species (chemical_reaction_network.species)"
-                "'d':species (chemical_reaction_network.species), ")
+    # Create the sbml_reaction in SBML
+    sbml_reaction = model.createReaction()
+    all_ids = getAllIds(model.getSBMLDocument().getListOfAllElements())
+    trans = SetIdFromNames(all_ids)
+    sbml_reaction.setId(trans.getValidIdForName(reaction_id))
+    sbml_reaction.setName(sbml_reaction.getId())
 
-        s = str(propensity_params['s1']).replace("'", "")
-        d = str(propensity_params['d']).replace("'", "")
-        s_species_id = getSpeciesByName(model,s).getId()
-        d_species_id = getSpeciesByName(model,d).getId()
+    # Create the kinetic law and corresponding local propensity parameters
+    crn_reaction.propensity_type.create_kinetic_law(model=model,
+                                                    sbml_reaction=sbml_reaction,
+                                                    stochastic=stochastic,
+                                                    crn_reaction=crn_reaction,
+                                                    reverse_reaction=reverse_reaction,
+                                                    **kwargs)
 
-        ratestring+=f"*{d_species_id}/({s_species_id}^n+K)"
+    # Create the reactants and products for the sbml_reaction
```

Comment:
@WilliamIX , if you want to set up the reverse reaction as two separate reactions, you need to flip the reactant/product setup:
```python
if reverse_reaction:
    _create_reactants(reactant_list=crn_reaction.outputs, sbml_reaction=sbml_reaction, model=model)
    _create_products(product_list=crn_reaction.inputs, sbml_reaction=sbml_reaction, model=model)
```

Reply:
Done